src/dense.jl

```diff
@@ -0,0 +1,243 @@
+
+# Constructor
+
+    AutoFiniteDifferences(; fdm)
+"""
+Base.@kwdef struct AutoFiniteDifferences{T} <: AbstractADType
+    fdm::T
+end
+
+mode(::AutoFiniteDifferences) = FiniteDifferencesMode()
+
+"""
+    AutoForwardDiff{chunksize,T}
+
+Chooses [ForwardDiff.jl](https://github.com/JuliaDiff/ForwardDiff.jl).
+
+# Type parameters
+
+- `chunksize`: the preferred [chunk size](https://juliadiff.org/ForwardDiff.jl/stable/user/advanced/#Configuring-Chunk-Size) to evaluate several derivatives at once
+
+# Fields
+
+- `tag::T`: a [custom tag](https://juliadiff.org/ForwardDiff.jl/release-0.10/user/advanced.html#Custom-tags-and-tag-checking-1) to handle nested differentiation calls (usually not necessary)
+
+# Constructors
+
+    AutoForwardDiff(; chunksize=nothing, tag=nothing)
+"""
+struct AutoForwardDiff{chunksize, T} <: AbstractADType
+    tag::T
+end
+
+function AutoForwardDiff(; chunksize = nothing, tag = nothing)
+    AutoForwardDiff{chunksize, typeof(tag)}(tag)
+end
+
+mode(::AutoForwardDiff) = ForwardMode()
+
+"""
+    AutoModelingToolkit
+
+Chooses [ModelingToolkit.jl](https://github.com/SciML/ModelingToolkit.jl).
+
+# Fields
+
+- `obj_sparse::Bool`
+- `cons_sparse::Bool`
+
+# Constructor
+
+    AutoModelingToolkit(; obj_sparse=false, cons_sparse=false)
+"""
+Base.@kwdef struct AutoModelingToolkit <: AbstractADType
+    obj_sparse::Bool = false
+    cons_sparse::Bool = false
+end
+
+mode(::AutoModelingToolkit) = SymbolicMode()
+
+"""
+    AutoPolyesterForwardDiff{chunksize,T}
+
+Chooses [PolyesterForwardDiff.jl](https://github.com/JuliaDiff/PolyesterForwardDiff.jl).
+
+# Type parameters
+
+- `chunksize`: the preferred [chunk size](https://juliadiff.org/ForwardDiff.jl/stable/user/advanced/#Configuring-Chunk-Size) to evaluate several derivatives at once
+
+# Fields
+
+- `tag::T`: a [custom tag](https://juliadiff.org/ForwardDiff.jl/release-0.10/user/advanced.html#Custom-tags-and-tag-checking-1) to handle nested differentiation calls (usually not necessary)
+
+# Constructors
+
+    AutoPolyesterForwardDiff(; chunksize=nothing, tag=nothing)
+"""
+struct AutoPolyesterForwardDiff{chunksize, T} <: AbstractADType
+    tag::T
+end
+
+function AutoPolyesterForwardDiff(; chunksize = nothing, tag = nothing)
+    AutoPolyesterForwardDiff{chunksize, typeof(tag)}(tag)
+end
+
+mode(::AutoPolyesterForwardDiff) = ForwardMode()
+
+"""
+    AutoReverseDiff
+
+Chooses [ReverseDiff.jl](https://github.com/JuliaDiff/ReverseDiff.jl).
+
+# Fields
+
+- `compile::Bool`: whether to [compile the tape](https://juliadiff.org/ReverseDiff.jl/api/#ReverseDiff.compile) prior to differentiation
+
+# Constructor
+
+    AutoReverseDiff(; compile=false)
+"""
+Base.@kwdef struct AutoReverseDiff <: AbstractADType
+    compile::Bool = false
+end
+
+mode(::AutoReverseDiff) = ReverseMode()
+
+"""
+    AutoTapir
+
+Chooses [Tapir.jl](https://github.com/withbayes/Tapir.jl).
+
+!!! danger
+    This package is experimental, use at your own risk.
```

Comment:
We should let Tapir say that, otherwise this will linger and potentially not be updated for years 😅

Reply:
Fair enough ^^

Reply:
removing the warning